### ISSUE TITLE
kmrcl no longer compiles: BIR verification failed.

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#1224]
reduced to
```lisp
Starting cclasp-boehmprecise-0.4.4-1094-g7e4aed1d3-cst ... loading image...
Writing jitted symbols to /tmp/perf-23212.map
Top level in: #<PROCESS TOP-LEVEL @0x10ccd88f1 (Running)>.
COMMON-LISP-USER> (defun last-cHar (s)
  (declare (simple-string s))
  (when (stringp s)
    (let ((len (length s)))
      (when (plusp len))
      (schar s (1- len)))))
; caught ERROR:
;   
;   -------module-------
;   constants: (ROW-MAJOR-AREF NIL 0 STRINGP CORE:TWO-ARG-- 1)
;   function LAST-CHAR (s) 
;        with environment () 
;        with start iblock LAST-CHAR-START
;     iblock LAST-CHAR-START ():
;      dynenv = (FUNCTION LAST-CHAR)
;        (leti s) -> S
;        (constant-reference 'stringp) -> 26 ; ctype: (VALUES (MEMBER STRINGP)
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (primop 26 fdefinition) -> STRINGP  ; ctype: (VALUES &OPTIONAL &REST T)
;        (readvar s) -> 27                   ; ctype: (VALUES T
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (thei 27
;    (values (or (simple-array base-char (*)) (simple-array character (*)))
;            &optional
;            &rest
;            nil)
;    nil) -> 28                              ; ctype: (VALUES
;                                                      (OR
;                                                       (SIMPLE-ARRAY BASE-CHAR
;                                                        (*))
;                                                       (SIMPLE-ARRAY CHARACTER
;                                                        (*)))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (call stringp 28) -> 29             ; ctype: (VALUES &OPTIONAL &REST T)
;        (cast 29 multiple-values (object)) -> 30 ; ctype: (VALUES &OPTIONAL
;                                                                  &REST
;                                                                  T)
;        (thei 30 (values (member nil t) &optional &rest nil) nil) -> 31 ; ctype: (VALUES
;                                                                                  (MEMBER
;                                                                                   NIL
;                                                                                   T)
;                                                                                  &OPTIONAL
;                                                                                  &REST
;                                                                                  NIL)
;        (ifi 31 (if-then if-else))
;     iblock IF-ELSE ():
;      dynenv = (FUNCTION LAST-CHAR)
;        (constant-reference 'nil) -> 24     ; ctype: (VALUES (MEMBER NIL)
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (cast 24 (object) multiple-values) -> 25 ; ctype: (VALUES (MEMBER NIL)
;                                                                  &OPTIONAL
;                                                                  &REST
;                                                                  NIL)
;        (jump 25 (merge)) -> 2              ; ctype: (VALUES &OPTIONAL
;                                                             (OR (MEMBER NIL)
;                                                                 CHARACTER)
;                                                             &REST
;                                                             NIL)
;     iblock IF-THEN ():
;      dynenv = (FUNCTION LAST-CHAR)
;        (readvar s) -> 3                    ; ctype: (VALUES T
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (thei 3
;    (values (or (simple-array base-char (*)) (simple-array character (*)))
;            &optional
;            &rest
;            nil)
;    nil) -> 4                               ; ctype: (VALUES
;                                                      (OR
;                                                       (SIMPLE-ARRAY BASE-CHAR
;                                                        (*))
;                                                       (SIMPLE-ARRAY CHARACTER
;                                                        (*)))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (primop 4 vector-length) -> 5       ; ctype: (VALUES &OPTIONAL &REST T)
;        (thei 5 (values (integer 0 (2305843009213693951)) &optional &rest nil) trusted) -> 6 ; ctype: (VALUES
;                                                                                                       (INTEGER
;                                                                                                        0
;                                                                                                        (2305843009213693951))
;                                                                                                       &OPTIONAL
;                                                                                                       &REST
;                                                                                                       NIL)
;        (leti 6) -> LEN
;        (readvar len) -> 7                  ; ctype: (VALUES
;                                                      (INTEGER 0
;                                                       (2305843009213693951))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (constant-reference '0) -> 8        ; ctype: (VALUES (MEMBER 0)
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (primop 7 8 two-arg-fixnum->) -> 9  ; ctype: (VALUES &OPTIONAL &REST T)
;        (cast 9 (object) nil) -> 10         ; ctype: (VALUES &OPTIONAL &REST T)
;        (readvar s) -> 11                   ; ctype: (VALUES T
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (thei 11
;    (values (or (simple-array base-char (*)) (simple-array character (*)))
;            &optional
;            &rest
;            nil)
;    nil) -> 12                              ; ctype: (VALUES
;                                                      (OR
;                                                       (SIMPLE-ARRAY BASE-CHAR
;                                                        (*))
;                                                       (SIMPLE-ARRAY CHARACTER
;                                                        (*)))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (constant-reference 'two-arg--) -> 13 ; ctype: (VALUES
;                                                        (MEMBER TWO-ARG--)
;                                                        &OPTIONAL
;                                                        &REST
;                                                        NIL)
;        (primop 13 fdefinition) -> TWO-ARG-- ; ctype: (VALUES &OPTIONAL
;                                                              &REST
;                                                              T)
;        (readvar len) -> 14                 ; ctype: (VALUES
;                                                      (INTEGER 0
;                                                       (2305843009213693951))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (constant-reference '1) -> 15       ; ctype: (VALUES (MEMBER 1)
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (call two-arg-- 14 15) -> 16        ; ctype: (VALUES &OPTIONAL &REST T)
;        (cast 16 multiple-values (object)) -> 17 ; ctype: (VALUES &OPTIONAL
;                                                                  &REST
;                                                                  T)
;        (thei 17 (values number &optional &rest nil) nil) -> 18 ; ctype: (VALUES
;                                                                          NUMBER
;                                                                          &OPTIONAL
;                                                                          &REST
;                                                                          NIL)
;        (thei 18 (values (integer 0 2305843009213693950) &optional &rest t) external) -> 19 ; ctype: (VALUES
;                                                                                                      NUMBER
;                                                                                                      &OPTIONAL
;                                                                                                      &REST
;                                                                                                      NIL)
;        (constant-reference 'row-major-aref) -> 20 ; ctype: (VALUES
;                                                             (MEMBER
;                                                              ROW-MAJOR-AREF)
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (primop 20 fdefinition) -> ROW-MAJOR-AREF ; ctype: (VALUES &OPTIONAL
;                                                                   &REST
;                                                                   T)
;        (thei 19 (values (integer 0 (2305843009213693950)) &optional &rest t) external) -> 21 ; ctype: (VALUES
;                                                                                                        NUMBER
;                                                                                                        &OPTIONAL
;                                                                                                        &REST
;                                                                                                        NIL)
;        (call row-major-aref 12 21) -> 22   ; ctype: (VALUES T
;                                                             &OPTIONAL
;                                                             &REST
;                                                             NIL)
;        (thei 22 (values character &optional &rest t) nil) -> 23 ; ctype: (VALUES
;                                                                           CHARACTER
;                                                                           &OPTIONAL
;                                                                           &REST
;                                                                           NIL)
;        (jump 23 (merge)) -> 2              ; ctype: (VALUES &OPTIONAL
;                                                             (OR (MEMBER NIL)
;                                                                 CHARACTER)
;                                                             &REST
;                                                             NIL)
;     iblock MERGE (2):
;      dynenv = (FUNCTION LAST-CHAR)
;        (returni 2)
;   function (LAMBDA ()) () 
;        with environment () 
;        with start iblock (LAMBDA ())-START
;     iblock (LAMBDA ())-START ():
;      dynenv = (FUNCTION (LAMBDA ()))
;        (enclose last-char) -> 0            ; ctype: (VALUES
;                                                      (FUNCTION
;                                                       (&OPTIONAL &REST T)
;                                                       (VALUES &OPTIONAL
;                                                               &REST
;                                                               T))
;                                                      &OPTIONAL
;                                                      &REST
;                                                      NIL)
;        (cast 0 (object) multiple-values) -> 1 ; ctype: (VALUES
;                                                         (FUNCTION
;                                                          (&OPTIONAL &REST T)
;                                                          (VALUES &OPTIONAL
;                                                                  &REST
;                                                                  T))
;                                                         &OPTIONAL
;                                                         &REST
;                                                         NIL)
;        (returni 1)
;   BIR verification failed. The IR is in an inconsistent state.
;   This probably indicates a problem in the compiler; please report it.
;   
;   Problems pertaining to function #<FUNCTION LAST-CHAR>:
;     
;          (primop 7 8 two-arg-fixnum->) -> 9
;     is used by #<CLASP-CLEAVIR-BMIR:CAST>, not an ifi instruction
;     
;     at unknown location
; 

Condition of type: VERIFICATION-FAILED
-------module-------
constants: (ROW-MAJOR-AREF NIL 0 STRINGP CORE:TWO-ARG-- 1)
function LAST-CHAR (s) 
     with environment #1=() 
     with start iblock LAST-CHAR-START
  iblock LAST-CHAR-START #1#:
   dynenv = (FUNCTION LAST-CHAR)
     (leti s) -> S
     (constant-reference 'stringp) -> 26     ; ctype: (VALUES (MEMBER STRINGP) &OPTIONAL &REST NIL)
     (primop 26 fdefinition) -> STRINGP      ; ctype: #2=(VALUES &OPTIONAL &REST T)
     (readvar s) -> 27                       ; ctype: (VALUES T &OPTIONAL &REST NIL)
     (thei 27 (values #3=(or (simple-array base-char (*)) (simple-array character (*))) &optional &rest nil) nil) -> 28 ; ctype: (VALUES #3# &OPTIONAL &REST NIL)
     (call stringp 28) -> 29                 ; ctype: #2#
     (cast 29 multiple-values #4=(object)) -> 30 ; ctype: #2#
     (thei 30 (values #5=(member nil t) &optional &rest nil) nil) -> 31 ; ctype: (VALUES #5# &OPTIONAL &REST NIL)
     (ifi 31 (if-then if-else))
  iblock IF-ELSE #1#:
   dynenv = (FUNCTION LAST-CHAR)
     (constant-reference 'nil) -> 24         ; ctype: #6=(VALUES #7=(MEMBER NIL) &OPTIONAL &REST NIL)
     (cast 24 #8=(object) multiple-values) -> 25 ; ctype: #6#
     (jump 25 (merge)) -> 2                  ; ctype: #9=(VALUES &OPTIONAL (OR #7# CHARACTER) &REST NIL)
  iblock IF-THEN #1#:
   dynenv = (FUNCTION LAST-CHAR)
     (readvar s) -> 3                        ; ctype: (VALUES T &OPTIONAL &REST NIL)
     (thei 3 (values #3# &optional &rest nil) nil) -> 4 ; ctype: (VALUES #3# &OPTIONAL &REST NIL)
     (primop 4 vector-length) -> 5           ; ctype: #2#
     (thei 5 (values #10=(integer 0 (2305843009213693951)) &optional &rest nil) trusted) -> 6 ; ctype: (VALUES #10# &OPTIONAL &REST NIL)
     (leti 6) -> LEN
     (readvar len) -> 7                      ; ctype: (VALUES #10# &OPTIONAL &REST NIL)
     (constant-reference '0) -> 8            ; ctype: (VALUES (MEMBER 0) &OPTIONAL &REST NIL)
     (primop 7 8 two-arg-fixnum->) -> 9      ; ctype: #11=(VALUES &OPTIONAL &REST T)
     (cast 9 (object) nil) -> 10             ; ctype: #11#
     (readvar s) -> 11                       ; ctype: (VALUES T &OPTIONAL &REST NIL)
     (thei 11 (values #3# &optional &rest nil) nil) -> 12 ; ctype: (VALUES #3# &OPTIONAL &REST NIL)
     (constant-reference 'two-arg--) -> 13   ; ctype: (VALUES (MEMBER TWO-ARG--) &OPTIONAL &REST NIL)
     (primop 13 fdefinition) -> TWO-ARG--    ; ctype: #2#
     (readvar len) -> 14                     ; ctype: (VALUES #10# &OPTIONAL &REST NIL)
     (constant-reference '1) -> 15           ; ctype: (VALUES (MEMBER 1) &OPTIONAL &REST NIL)
     (call two-arg-- 14 15) -> 16            ; ctype: #2#
     (cast 16 multiple-values #4#) -> 17     ; ctype: #2#
     (thei 17 (values number &optional &rest nil) nil) -> 18 ; ctype: (VALUES NUMBER &OPTIONAL &REST NIL)
     (thei 18 (values (integer 0 2305843009213693950) &optional &rest t) external) -> 19 ; ctype: (VALUES NUMBER &OPTIONAL &REST NIL)
     (constant-reference 'row-major-aref) -> 20 ; ctype: (VALUES (MEMBER ROW-MAJOR-AREF) &OPTIONAL &REST NIL)
     (primop 20 fdefinition) -> ROW-MAJOR-AREF ; ctype: #2#
     (thei 19 (values (integer 0 (2305843009213693950)) &optional &rest t) external) -> 21 ; ctype: (VALUES NUMBER &OPTIONAL &REST NIL)
     (call row-major-aref 12 21) -> 22       ; ctype: (VALUES T &OPTIONAL &REST NIL)
     (thei 22 (values character &optional &rest t) nil) -> 23 ; ctype: (VALUES CHARACTER &OPTIONAL &REST NIL)
     (jump 23 (merge)) -> 2                  ; ctype: #9#
  iblock MERGE (2):
   dynenv = (FUNCTION LAST-CHAR)
     (returni 2)
function (LAMBDA NIL) #1# 
     with environment #1# 
     with start iblock (LAMBDA ())-START
  iblock (LAMBDA ())-START #1#:
   dynenv = (FUNCTION (LAMBDA NIL))
     (enclose last-char) -> 0                ; ctype: #12=(VALUES (FUNCTION (&OPTIONAL &REST T) (VALUES &OPTIONAL &REST T)) &OPTIONAL &REST NIL)
     (cast 0 #8# multiple-values) -> 1       ; ctype: #12#
     (returni 1)
BIR verification failed. The IR is in an inconsistent state.
This probably indicates a problem in the compiler; please report it.
Problems pertaining to function #<FUNCTION LAST-CHAR>:
  
       (primop 7 8 two-arg-fixnum->) -> 9
  is used by #<CLASP-CLEAVIR-BMIR:CAST>, not an ifi instruction
  
Available restarts:
(use :r1 to invoke restart 1, etc.)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.


((METHOD CLEAVIR-BIR:VERIFY (PROGN) (CLEAVIR-BIR:MODULE)) #<CLEAVIR-BIR:MODULE>)
COMMON-LISP-USER>> 

````

**Comment by Bike** [comment]
Should be fixed by beea91c, but I have only tested the example.

**Comment by kpoeck** [comment]
(ql:quickload :kmrcl) works again, so fixed
